ResearchKit/Consent/ORKVisualConsentStepViewController.m

```diff
@@ -251,10 +252,30 @@ - (IBAction)goToPreviousPage {
 }
 
 - (IBAction)next {
-    [self showViewController:[self viewControllerForIndex:[self currentIndex]+1] forward:YES animated:YES];
-    ORKAccessibilityPostNotificationAfterDelay(UIAccessibilityScreenChangedNotification, nil, 0.5);
+    ORKConsentSceneViewController *currentConsentSceneViewController = [self viewControllerForIndex:[self currentIndex]];
+    [currentConsentSceneViewController setScrollEnabled:NO];
+    if ([currentConsentSceneViewController isScrolledToTop])
+    {
+        [self showNextViewController];
+    }
+    else
+    {
+        [currentConsentSceneViewController setScrollViewDelegate:self];
+        [currentConsentSceneViewController scrollToTopAnimated:YES];
+    }
 }
 
+- (void)scrollViewDidEndScrollingAnimation:(UIScrollView *)scrollView {
+    self.delegate = nil;
```

Comment:
Oops, you are right. ;)

It won't be needed when we switch to the block callback approach.